File: Find the Difference/Find_the_Difference.cpp
// Runtime: 4 ms, faster than 100.00% of C++ online submissions for Find the Difference.
// Memory Usage: 9.1 MB, less than 93.39% of C++ online submissions for Find the Difference.

class Solution 
{
public:
    char findTheDifference(string s, string t) 
    {
        vector<int> vec(26, 0);
        int i;
        for (i = 0; i < s.length(); i++)
            vec[s[i] - 'a']++;
        for (i = 0; i < t.length(); i++)
            if (--vec[t[i] - 'a'] == -1)
                break;
        return t[i];
    }
};

File: Find the Difference/Find_the_Difference.py
# Runtime: 60 ms, faster than 13.36% of Python3 online submissions for Find the Difference.
# Memory Usage: 13.3 MB, less than 5.00% of Python3 online submissions for Find the Difference.

class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        vec = [0 for i in range(26)]

        for char in s:
            vec[ord(char) - 97] += 1
        
        for char in t:
            vec[ord(char) - 97] -= 1
            if vec[ord(char) - 97] == -1:
                return char